python/python/benchmarks/test_search.py

```diff
@@ -54,8 +54,18 @@ def create_base_dataset(data_dir: Path) -> lance.LanceDataset:
     if dataset:
         return dataset
 
-    table = create_table(NUM_ROWS, offset=0)
-    dataset = lance.write_dataset(table, tmp_path)
+    rows_remaining = NUM_ROWS
+    offset = 0
+    dataset = None
+    while rows_remaining > 0:
+        next_batch_length = min(rows_remaining, 1024 * 1024)
+        rows_remaining -= next_batch_length
+        table = create_table(next_batch_length, offset)
+        if offset == 0:
+            dataset = lance.write_dataset(table, tmp_path)
+        else:
+            dataset = lance.write_dataset(table, tmp_path, mode="append")
+        offset += next_batch_length
```

Comment:
This is not strictly necessary since the NUM_ROWS constant is 100K but I was doing some temporary testing at 50M rows and so needed it to be iterative.